server/public/model/client4.go

```diff
@@ -4608,37 +4608,42 @@ func (c *Client4) GenerateSupportPacket(ctx context.Context) ([]byte, *Response,
 
 // GetPing will return ok if the running goRoutines are below the threshold and unhealthy for above.
 func (c *Client4) GetPing(ctx context.Context) (string, *Response, error) {
-	r, err := c.DoAPIGet(ctx, c.systemRoute()+"/ping", "")
-	if r != nil && r.StatusCode == 500 {
-		defer r.Body.Close()
-		return StatusUnhealthy, BuildResponse(r), err
-	}
-	if err != nil {
-		return "", BuildResponse(r), err
+	ping, resp, err := c.GetPingWithOptions(ctx, SystemPingOptions{})
+	status := ""
+	if ping != nil {
+		status = ping["status"]
 	}
-	defer closeBody(r)
-	return MapFromJSON(r.Body)["status"], BuildResponse(r), nil
+	return status, resp, err
 }
 
 // GetPingWithServerStatus will return ok if several basic server health checks
 // all pass successfully.
 func (c *Client4) GetPingWithServerStatus(ctx context.Context) (string, *Response, error) {
-	r, err := c.DoAPIGet(ctx, c.systemRoute()+"/ping?get_server_status="+c.boolString(true), "")
-	if r != nil && r.StatusCode == 500 {
-		defer r.Body.Close()
-		return StatusUnhealthy, BuildResponse(r), err
+	ping, resp, err := c.GetPingWithOptions(ctx, SystemPingOptions{FullStatus: true})
+	status := ""
+	if ping != nil {
+		status = ping["status"]
 	}
-	if err != nil {
-		return "", BuildResponse(r), err
-	}
-	defer closeBody(r)
-	return MapFromJSON(r.Body)["status"], BuildResponse(r), nil
+	return status, resp, err
 }
 
 // GetPingWithFullServerStatus will return the full status if several basic server
 // health checks all pass successfully.
 func (c *Client4) GetPingWithFullServerStatus(ctx context.Context) (map[string]string, *Response, error) {
-	r, err := c.DoAPIGet(ctx, c.systemRoute()+"/ping?get_server_status="+c.boolString(true), "")
+	return c.GetPingWithOptions(ctx, SystemPingOptions{FullStatus: true})
+}
+
+// GetPingWithOptions will return the status according to the options
+func (c *Client4) GetPingWithOptions(ctx context.Context, options SystemPingOptions) (map[string]string, *Response, error) {
```

Comment:
We got four methods, all hitting the same endpoint. Maybe it's worth simplifying to only `GetPingWithOptions`.

Reply:
Yes, it's to keep backwards compatibility, the other methods already calling the new `GetPingWithOptions` method. In time we will probably delete others.

Reply:
1/5 that we then should add a deprecation warning to the methods that we want to remove.

Reply:
Right, added 👍